src/logic/Applications.swift

```diff
@@ -115,7 +124,13 @@ class Applications {
             // Here I consider there is an edge case where AltTab is starting up and this program has been loading, then it is possible that axUiElement.windows() will throw an exception.
             // I'm not quite sure if this happens, but even if it does, then after restarting this application, AltTab captures its window without any problem. I think this happens rarely.
             let allWindows = CGWindow.windows(.optionAll)
-            guard let winApp = (allWindows.first { app.processIdentifier == $0.ownerPID() && $0.isNotMenubarOrOthers() && $0.id() != nil && $0.title() != nil}) else {
+            guard let winApp = (allWindows.first { app.processIdentifier == $0.ownerPID()
+                    && $0.isNotMenubarOrOthers()
+                    && $0.id() != nil
+                    && $0.bounds() != nil
+                    && CGRect(dictionaryRepresentation: $0.bounds()!)!.width > 0
+                    && CGRect(dictionaryRepresentation: $0.bounds()!)!.height > 0
+            }) else {
```

Comment:
As I said in the other ticket, you're re-implementing a mirror to https://github.com/lwouis/alt-tab-macos/blob/b668928c89dd256b39084d47b523ea882ebf6833/src/api-wrappers/AXUIElement.swift#L76 here. This will not be good enough. AltTab will thus show windows it shouldn't show.

Reply:
I add these codes just to filter those apps which have no windows(e.g. com.apple.universalcontroller). This is for those apps launched before AltTab. At that time, we haven't go to `AXUIElement.isActualWindow`. The code is just there to allow us to avoid observing to applications we are not interested in and to avoid wasting CPU time on `subscribeToNotification` retries.